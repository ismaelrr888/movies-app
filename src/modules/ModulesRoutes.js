import Backdrop from "@material-ui/core/Backdrop";
import CircularProgress from "@material-ui/core/CircularProgress";
import { makeStyles } from "@material-ui/core/styles";
import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import {
  BrowserRouter as Router,
  Redirect,
  Route,
  Switch,
} from "react-router-dom";
import { checkingFinish, startChecking } from "../actions/auth";
import UserLayout from "../layout/UserLayout";
import LoginContainer from "../modules/auth/containers/LoginContainer";
import RegisterContainer from "../modules/auth/containers/RegisterContainer";
import { PrivateRoute } from "../router/PrivateRoute";

const useStyles = makeStyles((theme) => ({
  backdrop: {
    zIndex: theme.zIndex.drawer + 1,
    color: "#fff",
  },
}));

export const ModulesRoutes = () => {
  const classes = useStyles();
  const { checking, uid } = useSelector((state) => state.auth);
  const dispatch = useDispatch();

  useEffect(() => {
    if (localStorage.getItem("token")) {
      dispatch(startChecking());
    } else {
      dispatch(checkingFinish());
    }
  }, [dispatch]);

  return (
    <>
      <Backdrop className={classes.backdrop} open={checking}>
        <h5>Whait...</h5> <CircularProgress color="inherit" />
      </Backdrop>
      <Router>
        <div id="public-route">
          <Switch>
            <Route exact path="/" component={() => <Redirect to="/login" />} />
            <Route exact path="/register" component={RegisterContainer} />
            <Route exact path="/login" component={LoginContainer} />
            <PrivateRoute
              path="/user"
              component={UserLayout}
              isAuthenticated={!!uid}
            />
          </Switch>
        </div>
      </Router>
    </>
  );
};
