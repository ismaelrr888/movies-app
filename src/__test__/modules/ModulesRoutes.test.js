import "@testing-library/jest-dom";
import { mount } from "enzyme";
import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { ModulesRoutes } from "../../modules/ModulesRoutes";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe("Test in <ModulesRoutes />", () => {
  test("should show Whait...", () => {
    const initState = {
      auth: {
        loading: false,
        checking: true,
      },
    };

    const store = mockStore(initState);

    const wrapper = mount(
      <Provider store={store}>
        <ModulesRoutes />
      </Provider>
    );
    expect(wrapper.find("h5").exists()).toBe(true);
  });

  test("should show the public route", () => {
    const initState = {
      auth: {
        loading: false,
        checking: false,
        uid: null,
      },
    };

    const store = mockStore(initState);

    const wrapper = mount(
      <Provider store={store}>
        <ModulesRoutes />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("#public-route").exists()).toBe(true);
  });

  test("should show the private route", () => {
    const initState = {
      calendar: {
        events: [],
      },
      auth: {
        loading: false,
        checking: false,
        uid: "123",
        name: "ismael",
      },
    };

    const store = mockStore(initState);

    const wrapper = mount(
      <Provider store={store}>
        <ModulesRoutes />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("#movies-container").exists()).toBe(true);
  });
});
