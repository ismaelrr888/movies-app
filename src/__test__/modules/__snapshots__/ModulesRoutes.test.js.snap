// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <ModulesRoutes /> should show the public route 1`] = `
<div
  id="public-route"
>
  <main
    className="MuiContainer-root MuiContainer-maxWidthXs"
    id="container-login"
  >
    <div
      className="makeStyles-paper-1"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circular makeStyles-avatar-2 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
          />
        </svg>
      </div>
      <h1
        className="MuiTypography-root MuiTypography-h5"
      >
        Sign in
      </h1>
      <form
        className="makeStyles-form-3"
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink={false}
              >
                Email Address
                <span
                  aria-hidden={true}
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoComplete="email"
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  name="email"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="text"
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-7"
                  >
                    <span>
                      Email Address
                       *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink={false}
              >
                Password
                <span
                  aria-hidden={true}
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  name="password"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="password"
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-7"
                  >
                    <span>
                      Password
                       *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Sign In
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-flex-end"
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
              href="/register"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Don't have an account? Sign Up
            </a>
          </div>
        </div>
      </form>
    </div>
    <div
      className="MuiBox-root MuiBox-root-10"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
      >
        Copyright © 
        2021
        .
      </p>
    </div>
  </main>
</div>
`;
